hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/BlockContentsValidator.java

```diff
@@ -65,50 +68,106 @@ public void validateBlocks(@NonNull final List<Block> blocks) {
     }
 
     private static void validate(Block block) {
-        final var blockItems = block.items();
+        final var items = block.items();
+        if (items.isEmpty()) {
+            Assertions.fail("Block is empty");
+        }
 
-        // A block SHALL start with a `header`.
-        if (!blockItems.getFirst().hasBlockHeader()) {
-            Assertions.fail("Block does not start with a block header");
+        if (items.size() <= 2) {
+            Assertions.fail("Block contains only header and proof with no rounds");
         }
 
-        // A block SHALL end with a `state_proof`.
-        if (!blockItems.getLast().hasBlockProof()) {
-            Assertions.fail("Block does not end with a block proof");
+        // A block SHALL start with a `block_header`.
+        validateBlockHeader(items.getFirst());
+
+        validateRounds(items.subList(1, items.size() - 1));
```

Comment:
Not sure I understand this. We take the block, cut off the first and last block item, pass that sublist to `validateRounds(...)`, then assume each block item represents a single round?

```

    private static void validateRounds(final List<BlockItem> roundItems) {
        int currentIndex = 0;
        while (currentIndex < roundItems.size()) {
            currentIndex = validateSingleRound(roundItems, currentIndex);
        }
    }
```

Reply:
Ahh nevermind, `currentIndex` increments by X as parsed by `validateSingleRound(...)`, not by 1